### ISSUE TITLE
META Gnoweb UI Improvements

### ISSUE THREAD

**Issue by moul** [gnolang/gno#3355]
Centralizing here some feedback about the new Gnoweb UI.

**Comment by moul** [comment]
Add a way to easily see which network we are running on. This could be a ribbon or another indicator. 

We need distinct values for the portal, testnets, and local (gnodev).

Bonus: subtle custom CSS allows us to have a gentle background color to quickly identify the portal, local, and other options.

**Comment by moul** [comment]
![CleanShot 2024-12-17 at 17 19 47](https://github.com/user-attachments/assets/7205dd86-20dd-4513-9035-11deefc430d5)

Use `alt` or a similar key to identify the buttons when hovering the mouse over them.

**Comment by moul** [comment]
![CleanShot 2024-12-17 at 17 21 08](https://github.com/user-attachments/assets/5168b63c-bce0-4dab-9f16-20b1973cf2b5)

When empty, remove the block or add text explaining why it is empty.

**Comment by moul** [comment]
Strange vertical alignment
![CleanShot 2024-12-17 at 17 48 48](https://github.com/user-attachments/assets/d6f79941-f7c6-4716-a623-fe48e89622fa)

**Comment by moul** [comment]
bug breaking `txlink`:

https://github.com/user-attachments/assets/8437ab02-19ff-47ba-b867-ef480d3ec75f

**Comment by moul** [comment]
Lack of cookie for the key/address.

https://github.com/user-attachments/assets/8c127d2f-b38b-4e0b-b393-64d376dfcc59

**Comment by kazai777** [comment]
It would be nice to have dark mode

**Comment by leohhhn** [comment]
Generally very empty pages, for example here: https://gno.land/r/docs/hello

**Comment by stefann-01** [comment]
Naming auto-generated function calls for realms as 'docs' isn't very intuitive. It took me a few minutes to realize what it was. I initially thought it would lead to a general documentation page.

**Comment by leohhhn** [comment]
Inline code does not match font size of other text when in H1,H2, etc

<img width="1041" alt="Screenshot 2024-12-19 at 13 42 39" src="https://github.com/user-attachments/assets/b23ed0e4-e61a-464d-95dd-6e8ea98e2b05" />

**Comment by leohhhn** [comment]
1. Please add a back button!
2. We should make imports in the source tab clickable; depending on the domain set in the `gnoweb` config, the urls generated would differ:
 
if `gnoweb`'s config has `test5.gno.land` -> an import, such as "gno.land/r/leon/hof" would be clickable with the url `test5.gno.land/r/leon/hof`

**Comment by leohhhn** [comment]
It seems that with a `?query` inside of the path, the source/docs buttons don't work properly - clicking them adds to `$source` or `$help` to the url, but doesn't actually switch to the tab.

![Image](https://github.com/user-attachments/assets/6ea70b13-c60b-471e-98a2-4fedd80992cc)

**Comment by leohhhn** [comment]
Seems someone broke the TOC click to jump to anchor! Currently clicking just adds a single `#` to the url.

**Comment by leohhhn** [comment]
Seems strikethrough is also not rendered on `gnoweb` currently, similarly to how tables were disabled. Should we check for some other elements as well?

<img width="254" alt="Image" src="https://github.com/user-attachments/assets/dc3b6e7e-1dcd-43ec-9d98-aa4b9cb175de" />

**Comment by leohhhn** [comment]
In the docs page, when user inputs for functions contain `!`, and possibly `?`, generated commands sometimes omit a closing `"`, which makes them invalid. I remember in the old version of `gnoweb` that `!` was always escaped.

**Comment by matijamarjanovic** [comment]
Hey, this case should maybe be covered with appropriate message(when trying to access the "docs" tab for a package):

![Image](https://github.com/user-attachments/assets/9f53fad6-afdf-4593-8866-da6dd93c577a)

**Comment by leohhhn** [comment]
Let's add copy buttons to all rendered codeblocks. I see now downside to this. Any thoughts?

**Comment by leohhhn** [comment]
Would be cool to get some better data under the titles when searching on google. I assume this is something we could achieve with realm metadata, no? Anyways, even with no metadata, maybe we could set some default text?

![Image](https://github.com/user-attachments/assets/c753e6f4-4a2a-454b-9903-36786c4ccc57)

**Comment by alexiscolin** [comment]
Yeah I have an issue for that. I plan to tackle this meta data PR next week!👍

**Comment by alexiscolin** [comment]
@leohhhn cc https://github.com/gnolang/gno/issues/3333

**Comment by leohhhn** [comment]
This is an old meta issue I'm linking: https://github.com/gnolang/gno/issues/1187

Still hope we can do this easily: https://github.com/gnolang/gno/issues/3355#issuecomment-2643626201

Finally, to make things a bit nicer, when the header is also a link the ToC displays it like so:
<img width="410" height="171" alt="Image" src="https://github.com/user-attachments/assets/23f0ae2b-2cd6-4047-812c-90401df2f93c" />

I suggest we strip the link and just leave the link text in the ToC, while the content will still be linked properly.
